src/client.rs

```
@@ -121,7 +115,19 @@ async fn transfer(
     let new_conn = endpoint
         .connect(*remote, &host)?
         .await
-        .map_err(|e| anyhow!("failed to connect: {}", e))?;
+        .map_err(|e| {
+            if e == ConnectionError::TimedOut {
+                let socket = if cfg!(target_os = "windows") {
+                    std::net::UdpSocket::bind("0.0.0.0:0").unwrap()
+                } else {
+                    std::net::UdpSocket::bind("[::]:0").unwrap()
+                };
+                endpoint.rebind(socket)?;
+                Ok(())
```

You still need to return the error here.